libs/math/tests/unit/test_types.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "gtest/gtest.h"
+#include "math/tensor.hpp"
+
+namespace fetch {
+namespace math {
+namespace test {
+
+using FloatingTypes =
+    ::testing::Types<float, double, fetch::fixed_point::fp32_t, fetch::fixed_point::fp64_t>;
+using HighPrecisionFloatingTypes = ::testing::Types<float, double, fetch::fixed_point::fp64_t>;
+
+using IntAndFloatingTypes =
+    ::testing::Types<int32_t, int64_t, float, double, fetch::fixed_point::fp32_t,
+                     fetch::fixed_point::fp64_t>;
+using UnsignedIntAndFloatingTypes =
+    ::testing::Types<uint32_t, int32_t, uint64_t, int64_t, float, double,
+                     fetch::fixed_point::fp32_t, fetch::fixed_point::fp64_t>;
```

Comment:
Judging by the name I was surprised that this contained int32_t. Maybe we should call it FloatIntAndUIntTypes?